client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -248,17 +249,22 @@ private long addToInflight(PendingEvent event) {
          */
         private List<PendingEvent> removeInflightBelow(long ackLevel) {
             synchronized (lock) {
-                ConcurrentNavigableMap<Long, PendingEvent> acked = inflight.headMap(ackLevel, true);
-                List<PendingEvent> result = new ArrayList<>(acked.values());
-                acked.clear();
+                List<PendingEvent> result = new ArrayList<>();
+                Entry<Long, PendingEvent> entry = inflight.peekFirst();
+                while (entry != null && entry.getKey() <= ackLevel) {
```

Comment:
Can this be rewritten as?

```
Entry<Long, PendingEvent> entry;
while(!inflight.isEmpty() && (entry = inflight.peekFirst()) != null) {
    result.add(entry.getValue);
    inflight.pollFirst();
}
```

The goal here is to reduce the number of times `peekFirst` appears in the code.

Reply:
This ignores the loop conditional: `entry.getKey() <= ackLevel`